File: frontend/src/types/reward.ts
export interface WalletBalance {
  coin_balance: number;
  updated_at: string;
}

export interface Reward {
  id: string;
  name: string;
  description: string;
  coin_cost: number;
  reward_type: string;
  reward_value: string;
  is_active: boolean;
}

export interface RewardCatalogue {
  items: Reward[];
}

export interface Redemption {
  redemption_id: string;
  reward_id: string;
  reward_name: string;
  coins_spent: number;
  remaining_balance: number;
  status: string;
  redeemed_at: string;
}

File: backend/app/schemas/reward.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel


class WalletBalanceResponse(BaseModel):
    coin_balance: int
    updated_at: datetime


class RewardResponse(BaseModel):
    id: UUID
    name: str
    description: str
    coin_cost: int
    reward_type: str
    reward_value: Decimal
    is_active: bool


class RewardCatalogueResponse(BaseModel):
    items: list[RewardResponse]


class RedeemRewardRequest(BaseModel):
    reward_id: UUID


class RedemptionResponse(BaseModel):
    redemption_id: UUID
    reward_id: UUID
    reward_name: str
    coins_spent: int
    remaining_balance: int
    status: str
    redeemed_at: datetime